pkg/k8s/client.go

```diff
@@ -434,3 +441,21 @@ func (c *Client) GetTargetPortForServicePort(namespacedSvc types.NamespacedName,
 func IsHeadlessService(svc corev1.Service) bool {
 	return len(svc.Spec.ClusterIP) == 0 || svc.Spec.ClusterIP == corev1.ClusterIPNone
 }
+
+// GetMeshConfig returns the MeshConfig resource corresponding to the control plane
+func (c *Client) GetMeshConfig() configv1alpha2.MeshConfig {
+	var meshConfig configv1alpha2.MeshConfig
+	configs := c.informers.List(osminformers.InformerKeyMeshConfig)
+	if len(configs) != 1 {
+		log.Error().Msgf("Expected exactly one MeshConfig resource, got %d", len(configs))
+	}
```

Comment:
Can we keep this consistent with the previous behavior where we simply `GetByKey(mesh_config_name)` so we don't log errors on MeshConfigs not relevant to the controller if they exist on the cluster for whatever reason?

Reply:
the informer already has an object name field selector so it was redundant

```
istOption := configInformers.WithTweakListOptions(func(opt *metav1.ListOptions) {
	opt.FieldSelector = fields.OneTermEqualSelector(metav1.ObjectNameField, meshConfigName).String()
})
```

Reply:
In that case, just a GetByKey would suffice vs a List?

Reply:
sure will do